src/xo-app/vm/tab-disks.js

```diff
@@ -100,31 +96,62 @@ class NewDisk extends Component {
+      id => resourceSet => resourceSet.id === id
+    )
+  )
+
+  _getResolvedResourceSet = createSelector(
+    this._getResourceSet,
+    resolveResourceSet
+  )
+
+  _getResourceSetDiskLimit = createSelector(
+    this._getResourceSet,
+    resourceSet => get(resourceSet, 'limits.disk.available')
+  )
+
   render () {
-    const { vm } = this.props
+    const { vm, isAdmin } = this.props
     const { formatMessage } = this.props.intl
+    const { size } = this.state
+
+    const diskLimit = this._getResourceSetDiskLimit()
+    const resourceSet = this._getResolvedResourceSet()
+
+    const SelectSr_ = isAdmin || resourceSet == null ? SelectSr : SelectResourceSetsSr
 
     return <form id='newDiskForm'>
       <div className='form-group'>
-        <SelectSr predicate={this._getSrPredicate()} onChange={this._selectSr} required />
+        <SelectSr_
+          onChange={this.linkState('sr')}
+          predicate={this._getSrPredicate()}
+          required
+          resourceSet={isAdmin ? undefined : resourceSet}
+        />
       </div>
       <fieldset className='form-inline'>
         <div className='form-group'>
-          <input type='text' ref='name' placeholder={formatMessage(messages.vbdNamePlaceHolder)} className='form-control' required />
+          <input type='text' onChange={this.linkState('name')} placeholder={formatMessage(messages.vbdNamePlaceHolder)} className='form-control' required />
         </div>
         {' '}
         <div className='form-group'>
-          <SizeInput ref='size' placeholder={formatMessage(messages.vbdSizePlaceHolder)} required />
+          <SizeInput onChange={this.linkState('size')} placeholder={formatMessage(messages.vbdSizePlaceHolder)} required />
         </div>
         {' '}
         <div className='form-group'>
-          {vm.virtualizationMode === 'pv' && <span>{_('vbdBootable')} <Toggle ref='bootable' /> </span>}
-          <span>{_('vbdReadonly')} <Toggle ref='readOnly' /></span>
+          {vm.virtualizationMode === 'pv' && <span>{_('vbdBootable')} <Toggle onChange={this.toggleState('bootable')} /> </span>}
+          <span>{_('vbdReadonly')} <Toggle onChange={this.toggleState('readOnly')} /></span>
         </div>
         <span className='pull-right'>
-          <ActionButton form='newDiskForm' icon='add' btnStyle='primary' handler={this._createDisk}>{_('vbdCreate')}</ActionButton>
+          <ActionButton form='newDiskForm' icon='add' btnStyle='primary' handler={this._createDisk} disabled={diskLimit < size}>{_('vbdCreate')}</ActionButton>
         </span>
       </fieldset>
+      {resourceSet != null && diskLimit >= size && <em>{_('useQuotaWarning', { resourceSet: <strong>{resourceSet.name}</strong>, spaceLeft: formatSize(diskLimit) })}</em>}
+      {resourceSet != null && diskLimit < size && <em className='text-danger'>{_('notEnoughSpaceInResourceSet', { resourceSet: <strong>{resourceSet.name}</strong>, spaceLeft: formatSize(diskLimit) })}</em>}
```

Comment:
Merge those tests:

```js
{resourceSet != null && (
  diskLimit >= size
    ? // ...
    : // ...
}

Reply:
If `diskLimit` is undefined, I don't want anything to be shown, though.